criu/kerndat.c

```diff
@@ -1146,6 +1150,39 @@ int kerndat_has_nspid(void)
 	return ret;
 }
 
+static int kerndat_has_nftables_concat(void)
+{
+#if defined(CONFIG_HAS_NFTABLES_LIB_API_0) || defined(CONFIG_HAS_NFTABLES_LIB_API_1)
+	struct nft_ctx *nft;
+	int ret = 0;
+
+	nft = nft_ctx_new(NFT_CTX_DEFAULT);
```

Comment:
I can add it, but I think it wouldn't collide with anything, since no filtering rules are added (only a test set is created) which is added to separate CRIU table.
Having two instances of CRIU running at the same time would be a problem since the table name will for sure collide, I think actually that would also be the case during normal locking/unlocking like [here](https://github.com/checkpoint-restore/criu/pull/1539/files/1d7d0d252e350c0ec4f441c5b95dbb9ce6f99b02#diff-ffdccecb1d7573f5f13830e82a1b1c6e200f33b3bdb934ea9add47d68ad23406R3028). Is that a use case I should consider?

Reply:
It looks like the only difference between `add table inet CRIU` and `create table inet CRIU` is that the former will not return an error if the table already exists while `create` will return an error.

To avoid collisions, we could replace `add` with `create` and append the PID number specified with `-t` (`--tree`) to the table name.

Reply:
I believe that there is still a race window here... One kerndat_has_nftables_concat can remove the table under another kerndat_has_nftables_concat, and another would fail on no table and wrongly set kdat.has_nftables_concat = false and will save it to kerndat cache.

Actually running multiple migrations of different containers in parallel can be quite a usecase for cloud cluster load balancing.

Reply:
kerndat is cached, so it has to be collected once. But I agree that this looks like a race and we need to generate a unique name for this test table.

Reply:
Thank you for pointing this out. https://github.com/checkpoint-restore/criu/pull/1588